src/app/core/data-gateway/data-gateway.service.ts

```diff
@@ -51,13 +51,14 @@ export class DataGatewayService {
         return this.ipc.request("searchLocalProjects", {term, limit,}).toPromise();
     }
 
-    fetchFileContent(almostID: string, parse = false): Observable<string> {
+    fetchFileContent(almostID: string, parse = false, useSwap = true): Observable<string> {
 
         const source = DataGatewayService.getFileSource(almostID);
 
         if (source === "local") {
 
-            const fetch = Observable.empty().concat(this.ipc.request("getLocalFileContent", almostID)) as Observable<string>;
+            const fetch = useSwap ? Observable.empty().concat(this.ipc.request("getLocalFileContent", almostID)) as Observable<string> :
```

Comment:
it would be more readable if you write like:
const swap = useSwap ? Observable.empty() : this.updateSwap(almostID, null);
const fetch = Observable.empty().concat(swap, this.ipc.request("getLocalFileContent", almostID)) as Observable<string>;